modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3770,41 +3792,63 @@ def _groupby_shuffle(
                 by, agg_func, axis, groupby_kwargs, agg_args, agg_kwargs, how, drop
             )
 
-        if isinstance(by, type(self)) and drop:
-            by = by.columns.tolist()
+        if groupby_kwargs.get("level") is not None:
+            raise NotImplementedError(
```

Comment:
Just to be sure, these cases (`NotImplementedError`) will be executed on pure pandas?

Reply:
no, they will fall back to either [MapReduce](https://github.com/modin-project/modin/blob/4a32e71a7dd91d9ab7ab5d1763cc7fe0f52c08e4/modin/core/storage_formats/pandas/groupby.py#L96-L110) or [FullAxis](https://github.com/modin-project/modin/blob/4a32e71a7dd91d9ab7ab5d1763cc7fe0f52c08e4/modin/core/storage_formats/pandas/query_compiler.py#L3984-L4005) implementation